src/data-table/index.js

```diff
@@ -123,12 +125,19 @@ class DataTable extends Component {
 
     if (Array.isArray(columns) && columns.length > 0) {
       // apply default here since columns are not DataTableColumn instances
-      return columns.map(c => ({ ...columnDefaultProps, ...c }))
+      return columns.map(c => ({
+        ...columnDefaultProps,
+        sortType: (Number.isInteger(data[0][c.dataKey])) ? 'basic' : ((Number.isInteger(Date.parse(data[0][c.dataKey]))) ? 'date' : 'string'),
+        ...c,
+      }))
     }
 
     return (Array.isArray(children) ? children : [children])
       .filter(c => c.type === DataTableColumn || c.type.name === 'DataTableColumn')
-      .map(c => c.props)
+      .map(c => ({
+        ...c.props,
+        sortType: (Number.isInteger(data[0][c.dataKey])) ? 'basic' : ((Number.isInteger(Date.parse(data[0][c.dataKey]))) ? 'date' : 'string'),
```

Comment:
I think we should put this line before `...c.props,` so that we don't overwrite `sortType` in props passed in.